pkg/archive/buffer_stager.go

```diff
@@ -56,13 +57,15 @@ func NewBufferStager(sconfig *stage.StagerConfig, chkPt *audit.Checkpoint, schem
 	}
 	atomic.StoreUint64(&b.totalRows, uint64(totalRows))
 
-	b.chunker, err = table.NewChunker(sconfig.SrcTbl, sconfig.ChunkDuration, sconfig.Logger)
+	// Use slog.Default() for the chunker since spirit now uses slog.Logger
+	// Pass SrcTbl as both old and new table since we're archiving from the same table
+	b.chunker, err = table.NewChunker(sconfig.SrcTbl, sconfig.SrcTbl, sconfig.ChunkDuration, slog.Default())
```

Comment:
cc @morgo I am forced to pass a non nil table for destination here as the OpenAtWaterMark() for `optimisticChunker` [assumes](https://github.com/block/spirit/blob/aeb604d5ab50a6da1b409d9688c9bb1055b0b47f/pkg/table/chunker_optimistic.go#L209) that the destination table is not nil. As this is is archive phase (in which we copy all the rows from staged table without where clause, it uses optmistic chunker). I am not sure what the right thing to do here

Reply:
We have the same problem in move table operations. The API expects two tables because it uses this to calculate a column intersection -- but that doesn't make sense for all use cases.

I think what you have is fine for now. I might change the API in future to have something clearer for non-transforming cases.